rpc/eth_api.go

```diff
@@ -200,9 +207,52 @@ func (e *PublicEthAPI) Sign(address common.Address, data hexutil.Bytes) (hexutil
 }
 
 // SendTransaction sends an Ethereum transaction.
-func (e *PublicEthAPI) SendTransaction(args core.SendTxArgs) common.Hash {
-	var h common.Hash
-	return h
+func (e *PublicEthAPI) SendTransaction(args args.SendTxArgs) (common.Hash, error) {
+	// TODO: Change this functionality to find an unlocked account by address
+	if e.key == nil || !bytes.Equal(e.key.PubKey().Address().Bytes(), args.From.Bytes()) {
+		return common.Hash{}, keystore.ErrLocked
+	}
+
+	// Mutex lock the address' nonce to avoid assigning it to multiple requests
+	if args.Nonce == nil {
+		e.nonceLock.LockAddr(args.From)
+		defer e.nonceLock.UnlockAddr(args.From)
+	}
+
+	// Assemble transaction from fields
+	tx, err := types.GenerateFromArgs(args, e.cliCtx)
+	if err != nil {
+		return common.Hash{}, err
+	}
+
+	// ChainID must be set as flag to send transaction
+	chainID := viper.GetString(flags.FlagChainID)
+	// parse the chainID from a string to a base-10 integer
+	intChainID, ok := new(big.Int).SetString(chainID, 10)
+	if !ok {
+		return common.Hash{}, fmt.Errorf(
+			fmt.Sprintf("Invalid chainID: %s, must be integer format", chainID))
+	}
+
+	// Sign transaction
+	tx.Sign(intChainID, e.key.ToECDSA())
+
+	// Encode transaction by default Tx encoder
+	txEncoder := authutils.GetTxEncoder(e.cliCtx.Codec)
+	txBytes, err := txEncoder(tx)
+	if err != nil {
+		return common.Hash{}, err
+	}
+
+	// Broadcast transaction
+	res, err := e.cliCtx.BroadcastTx(txBytes)
+	// If error is encountered on the node, the broadcast will not return an error
+	fmt.Println(res.RawLog)
```

Comment:
keeping this?

Reply:
Yeah, at least for now. Will add a TODO to remove or explore later, but as of now there is no other way to check whether the tx fails or not because the err is if the broadcast fails, not the tx ante handler